spyder/plugins/editor/lsp/providers/window.py

```diff
@@ -0,0 +1,20 @@
+# -*- coding: utf-8 -*-
+
+# Copyright © Spyder Project Contributors
+# Licensed under the terms of the MIT License
+# (see spyder/__init__.py for details)
+
+"""Spyder Language Server Protocol Client window handler routines."""
+
+import logging
+
+from spyder.plugins.editor.lsp import LSPRequestTypes
+from spyder.plugins.editor.lsp.decorators import handles  # , send_request
+
+logger = logging.getLogger(__name__)
+
+
+class WindowProvider:
```

Comment:
Why is this class needed?

Reply:
LSP servers can return messages of different "scopes", one being `textDocument` (which in Spyder is handled by the `DocumentProvider` class).  In order to keep the message handlers separated by scope, I added a new class to handle the `window` scope.